blocks/library/gallery/index.js

```diff
@@ -131,23 +131,8 @@ export const settings = {
 					return files.length !== 1 && every( files, ( file ) => file.type.indexOf( 'image/' ) === 0 );
 				},
 				transform( files, onChange ) {
-					const block = createBlock( 'core/gallery', {
-						images: files.map( ( file ) => ( {
-							url: window.URL.createObjectURL( file ),
-						} ) ),
-					} );
-
-					Promise.all( files.map( ( file ) =>
-						createMediaFromFile( file )
-							.then( ( media ) => preloadImage( media.source_url ).then( () => media ) )
-					) ).then( ( medias ) => onChange( block.uid, {
-						images: medias.map( media => ( {
-							id: media.id,
-							url: media.source_url,
-							link: media.link,
-						} ) ),
-					} ) );
-
+					const block = createBlock( 'core/gallery' );
+					mediaUpload( files, ( images ) => onChange( block.uid, { images } ) );
```

Comment:
I think you need to rebase with master. I merged a change which adds 3rd param to `mediaUpload` function - media type. It was needed to allow upload of other types like audio or video.